programmers/다트-게임.js

```diff
@@ -0,0 +1,57 @@
+function solution(dartResult) {
+    let num = 0;
+    const answer = [];
+    let temp = 0;
+    
+
+    for (let i = 0; i < dartResult.length; i++) {
+        if (dartResult[i] >= 0 && dartResult[i] <= 9) {
+            if (dartResult[i] == 1 && dartResult[i + 1] == 0) {
+                temp = 10;
+                i++;
+            }
+            else {
+                temp = dartResult[i];
+            }
+        }
+        else if(dartResult[i] === 'S') {
+            answer.push(temp);
+        }
+        else if(dartResult[i] === 'D') {
+            answer.push(Math.pow(temp, 2));
+        }
+        else if(dartResult[i] === 'T'){
+            answer.push(Math.pow(temp, 3));
+        }
+        else if(dartResult[i] == '#') {
+           answer[answer.length - 1] *= -1;
+        }
+        else if(dartResult[i] == '*') {
+            answer[answer.length - 1] *= 2;
+            answer[answer.length - 2] *= 2;
+        }
+    }
+    for (let i = 0; i < answer.length; i++) {
+        num += Number(answer[i]);
+    }
+    return num;
```

Comment:
```suggestion
  return answer.reduce((firstScore, secondScore) => firstScore + secondScore);
```
reduce를 사용하면 이런식으로 표현도 가능하답니다
제가 볼땐 Number를 안해도 될꺼같은데 Number로 해준 이유가 궁금하네요